libcalico-go/lib/backend/k8s/conversion/conversion.go

```diff
@@ -262,12 +263,20 @@ func (c converter) K8sNetworkPolicyToCalico(np *networkingv1.NetworkPolicy) (*mo
 	// This order might change in future.
 	order := float64(1000.0)
 
+	var convErr cerrors.ErrorPolicyConversion
+	var convErrRules []cerrors.ErrorPolicyConversionRule
+
 	// Generate the ingress rules list.
 	var ingressRules []apiv3.Rule
 	for _, r := range np.Spec.Ingress {
 		rules, err := c.k8sRuleToCalico(r.From, r.Ports, np.Namespace, true)
 		if err != nil {
 			log.WithError(err).Warn("dropping k8s rule that couldn't be converted.")
+			// Add rule to conversion error slice
+			convErrRules = append(convErrRules, cerrors.ErrorPolicyConversionRule{
```

Comment:
One idea that you can implement or not depending on if you like it - it might be nice to encapsulate this a bit more into the error type to build a cleaner interface. For example:


```go

// A new type that encapsulates all the logic to track intermediate errors, and also
// build a summarized error to return if any are encountered.
errorTracker := cerrors.NetworkPolicyErrorTracker{}

...

if err != nil {
  errorTracker.BadEgressRule(r, err)
}

...

return kvp, errorTracker.Error()

```

Should make the code changes in this file a bit more streamlined.

Reply:
hm... yeah, this would be better, thanks @caseydavenport! will do!